core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -1052,6 +1053,41 @@ void testAggregateWithAlias(SqlExplainFormat format) {
     assertThat(result, isLinux(expected));
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5607">[CALCITE-5607]</a>
+   *
+   * <p>Before the fix, RelJson.toRex would throw an ArrayIndexOutOfBounds error
+   * when deserialization.
+   * */
+  @Test void testDeserializeMinusDateOperator() {
```

Comment:
The commit message and test case don't seem to align. This checks that `MINUS` can be deserialized but I don't see anything to suggest we've changed how the return `type` is being serialized.

Reply:
Yeah let me update the JIRA summary and the ticket title for this.
We're changing it to only affect MINUS